github/repos_test.go

```diff
@@ -679,3 +697,27 @@ func TestRepositoriesService_ListRequiredStatusChecksContexts(t *testing.T) {
 		t.Errorf("Repositories.ListRequiredStatusChecksContexts returned %+v, want %+v", contexts, want)
 	}
 }
+
+func TestRepositoriesService_ListRequiredStatusChecksContexts_unprotectedBranch(t *testing.T) {
+	setup()
+	defer teardown()
+
+	mux.HandleFunc("/repos/o/r/branches/b/protection/required_status_checks/contexts", func(w http.ResponseWriter, r *http.Request) {
+		testMethod(t, r, "GET")
+		testHeader(t, r, "Accept", mediaTypeProtectedBranchesPreview)
+		w.WriteHeader(http.StatusNotFound)
+		fmt.Fprint(w, `{
+  "message": "Not Found",
+  "documentation_url": "https://developer.github.com/v3"
+}`)
+	})
+
+	contexts, _, err := client.Repositories.ListRequiredStatusChecksContexts(context.Background(), "o", "r", "b")
+	if err != nil {
+		t.Errorf("Repositories.ListRequiredStatusChecksContexts returned error: %v", err)
+	}
+
+	if contexts != nil {
```

Comment:
How do you feel about checking if `len(contexts) != 0` instead?

I'm not sure if we should verify whether the method returns `nil` slice or `[]string{}` empty slice. Do you think we should?